src/spanningtrees/kruskal.jl

```diff
@@ -35,7 +35,7 @@ function kruskal_mst end
 )
 
     edge_list = Vector{KruskalHeapEntry{T}}()
-    mst = Vector{Edge}()
+    mst = Vector{AbstractEdge}()
```

Comment:
This is going to get a bit more complicated. We'll want to type-parameterize the function (which, because it's a `@traitfn`, will mean using the "old-style" parameterization) on `U<:AbstractEdge` and call this a `Vector{U}()`.